src/cli/commands/force/lightning/local/device/list.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { flags, FlagsConfig, SfdxCommand } from '@salesforce/command';
+import { Logger, Messages, SfdxError } from '@salesforce/core';
+import chalk from 'chalk';
+import cli from 'cli-ux';
+import { AndroidVirtualDevice } from '../../../../../../common/AndroidTypes';
+import { AndroidSDKUtils } from '../../../../../../common/AndroidUtils';
+import { CommandLineUtils } from '../../../../../../common/Common';
+import { IOSSimulatorDevice } from '../../../../../../common/IOSTypes';
+import { XcodeUtils } from '../../../../../../common/IOSUtils';
+import Setup from '../setup';
+
+// Initialize Messages with the current plugin directory
+Messages.importMessagesDirectory(__dirname);
+
+// Load the specific messages for this file. Messages from @salesforce/command, @salesforce/core,
+// or any library that is using the messages framework can also be loaded this way.
+const messages = Messages.loadMessages('@salesforce/lwc-dev-mobile', 'device');
+
+export default class List extends Setup {
+    public static description = messages.getMessage('commandDescription');
+
+    public static readonly flagsConfig: FlagsConfig = {
+        platform: flags.string({
+            char: 'p',
+            description: messages.getMessage('platformFlagDescription'),
+            longDescription: messages.getMessage('platformFlagDescription'),
+            required: true
+        })
+    };
+
+    public examples = [
+        `sfdx force:lightning:local:device:list -p iOS`,
+        `sfdx force:lightning:local:device:list -p Android`
+    ];
+
+    public async run(): Promise<any> {
+        const platform = this.flags.platform;
+        this.logger.info(`Device List command invoked for ${platform}`);
+
+        return new Promise<any>((resolve, reject) => {
+            super
+                .run() // run setup first
```

Comment:
Hmmm.  In this new world, the error message we're getting back for e.g. not having the required `-p` option is very ugly, compared with what we get back for the same scenario running the `setup` command directly.  I'm seeing if I can figure out why.

Reply:
The issue lies in some aspect of calling `super.run()` within the child (e.g. `List`) plugin.  If I extract the implementation of `Setup.run()` and call that directly from `List.run()`, instead of calling `super.run()`, it behaves as expected.

I think that maybe a base class is overly complicated in this case, for what it's going to have to be to make it work.  Directly calling `run()` seems (and has seemed) problematic for a number of reasons.

Let's do the following:

1. Remove the inheritance from `Setup` as a base class—let the plugin classes be siblings again.
2. Extract an accessible (public) helper method on `Setup`, that encapsulates the current implementation of `Setup.run()`.
3. Call that method in both `Setup.run()`, and the other plugins' `run()` implementations.

We're past due for making that modification anyway.

Reply:
Gah, never mind, the issue I was seeing was because I was using the `--json` flag in my tests.  Sorry.  Just disregard all of this.  It's working properly as is.